### PR TITLE
Add random walk features to atom-env

### PR DESCRIPTION
**Pull request by bengioe**
Adds:
- the ability compute random walk features of graph nodes, this is known to help GNNs' representational power (see [Dwivedi et al.](https://arxiv.org/abs/2110.07875)).
- these features by default to `MolBuildingEnvContext`, the atom environment context
- a docstring to `MolBuildingEnvContext.__init__`

### CODE REVIEW

**Comment by bearnshaw** [comment]
What are random walk features? Is there a reference you could share?

**Comment by bengioe** [comment]
Good point, as far as I know [this is where they were introduced](https://arxiv.org/abs/2110.07875), they also made their way into the recent SoTA [GraphGPS](https://arxiv.org/abs/2205.12454) paper.

File: src/gflownet/utils/graphs.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    else:
+        # P = D^-1 * A
+        P = torch.diag(deg_inv) @ to_dense_adj(g.edge_index, max_num_nodes=g.num_nodes)  # (1, num nodes, num nodes)
+    diags = []
+    if skip_odd:
+        Pmult = P @ P
+    else:
+        Pmult = P
+    Pk = Pmult
+    for k in range(k):
```
Comment (bearnshaw) [inline src/gflownet/utils/graphs.py]:
```suggestion
    for _ in range(k):
```
It's more pythonic to use `_` in a dummy loop.

**Review by bearnshaw — CHANGES_REQUESTED** [review]
Looks good, one requested change.

**Review by bearnshaw — APPROVED** [review]

**Pull request merged**